### PR TITLE
Add note on recommended way to manage JavaScript dependencies via NPM and link to Spago

### PR DESCRIPTION
**Pull request by mbbx6spp**
These suggested additions to the front page of the website are made based on discussions from work while onboarding developers new to Purescript and also via more constructive Twitter discussions.

I added instructions on installing the `purescript` compile via Homebrew and also via Nix. I maintain the Nix package right now and plan to update it when newer versions of Purescript are released. The Homebrew formula for purescript is also fairly current.

I added the final note about how to manage JS dependencies because that has come up in multiple conversations with people new to Purescript trying to get ramped up.

Happy to amend.

### CODE REVIEW

**Comment by natefaubion** [comment]
The current listed options are maintained by the core org. I don't have a problem with listing other options, but I would probably list them under a separate heading for third-party tools. @hdgarrood What do you think?

**Comment by natefaubion** [comment]
There is also this page, which is linked from the home page:
https://github.com/purescript/documentation/blob/master/ecosystem/Editor-and-tool-support.md

File: index.html [hunk 1]
```diff
@@ -193,11 +193,19 @@ <h3>npm</h3>
...
+          <p><code>nix-env -i purescript</code></p>
+          <p><small>(Install using Nix v2.3+ and a recent version of <code>&lt;nixpkgs&gt;</code>)</small></p>
+
           <h3>Tools</h3>
-          <p>The recommended build tool is Spago, which can be installed using <code>npm</code>:</p>
+          <p>The recommended build tool for Purescript is <a href="https://github.com/purescript/spago/blob/master/README.md" title="Spago: Purescrpt's package manager and build tool">Spago</a>, which can be installed using <code>npm</code>:</p>
           <p><code>npm install -g spago</code></p>
           <p>You might like to install some of these
             <a href="https://github.com/purescript/documentation/blob/master/ecosystem/Editor-and-tool-support.md">additional tools and editor plugins</a>.</p>
+          <p>NPM or Yarn is recommended for managing Javascript dependencies in your project.</p>
```
Comment (natefaubion) [inline index.html]: Note that recommending Yarn might prompt users to install purescript locally via yarn, which does not currently work due to PnP compatibility.

File: index.html [hunk 2]
```diff
@@ -193,11 +193,19 @@ <h3>npm</h3>
...
+          <h3>Homebrew</h3>
+          <p><code>brew install purescript</code><p>
+
+          <h3>Nix</h3>
+          <p><code>nix-env -i purescript</code></p>
+          <p><small>(Install using Nix v2.3+ and a recent version of <code>&lt;nixpkgs&gt;</code>)</small></p>
+
           <h3>Tools</h3>
-          <p>The recommended build tool is Spago, which can be installed using <code>npm</code>:</p>
+          <p>The recommended build tool for Purescript is <a href="https://github.com/purescript/spago/blob/master/README.md" title="Spago: Purescrpt's package manager and build tool">Spago</a>, which can be installed using <code>npm</code>:</p>
```
Comment (natefaubion) [inline index.html]:
```suggestion
          <p>The recommended build tool for PureScript is <a href="https://github.com/purescript/spago/blob/master/README.md" title="Spago: PureScript's package manager and build tool">Spago</a>, which can be installed using <code>npm</code>:</p>
```

File: index.html (hunk 1 above)
Comment (natefaubion) [inline index.html]:
```suggestion
          <p>NPM or Yarn is recommended for managing JavaScript dependencies in your project.</p>
```

Comment (mbbx6spp) [reply index.html]: I will amend, thanks.

**Comment by mbbx6spp** [comment]
I will make changes based on the feedback later this evening. Thank you much.

**Comment by mbbx6spp** [comment]
I linked a PR to the purescript/documentation repository where I will put the Homebrew and Nix installation methods in there. I do think it might be worthwhile keeping the note about using NPM to manage Javascript dependencies here though, but I will defer. Let me amend this PR now though.

**Comment by mbbx6spp** [comment]
Pushed updates to reflect the above and amended PR title to refect current changes.

**Pull request merged**